eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/Retry.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.exception.ErrorCondition;
+import com.azure.eventhubs.implementation.ClientConstants;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract class Retry {
+
+    public static final Retry NO_RETRY = new RetryExponential(Duration.ofSeconds(0), Duration.ofSeconds(0), 0);
+
+    private AtomicInteger retryCount = new AtomicInteger(0);
+
+    /**
+     * Check if the existing exception is a retryable exception.
+     *
+     * @param exception A exception that was observed for the operation to be retried.
+     * @return true if the exception is a retryable exception, otherwise false.
+     * @throws IllegalArgumentException when the exception is null.
+     */
+    public static boolean isRetryableException(Exception exception) {
+        if (exception == null) {
+            throw new IllegalArgumentException("exception cannot be null");
+        }
+
+        if (exception instanceof AmqpException) {
+            return ((AmqpException) exception).isTransient();
+        }
+        return false;
+    }
+
+    /**
+     * Get 'NO_RETRY' of current.
+     *
+     * @return Retry 'NO_RETRY'.
+     */
+    public static Retry getNoRetry() {
+        return Retry.NO_RETRY;
+    }
+
+    /**
+     * Get default configured Retry.
+     *
+     * @return Retry which has all default property set up.
+     */
+    public static Retry getDefault() {
+        return new RetryExponential(
+            ClientConstants.DEFAULT_RETRY_MIN_BACKOFF,
```

Comment:
Pull all the ClientConstants into this class. ClientConstants is a giant bag of random values, which makes it hard to reason about. If we pull these three constants into this class, we know that they definitely pertain to Retry.